### ISSUE TITLE
sql: test sql/gh-3199-no-mem-leaks.test.lua fails sometime

### ISSUE THREAD

**Issue by ImeevMA** [tarantool/tarantool#4198]
Tarantool version: 2.2.0-261-g55e1a14

OS version: Ubuntu 16.04

Bug description: Test sql/gh-3199-no-mem-leaks.test.lua fails sometime.

Steps to reproduce: Use any of these four reproducers:
```
- [sql/check-clear-ephemeral.test.lua, memtx]
- [sql/gh-3199-no-mem-leaks.test.lua, memtx]
```
```
- [sql/check-clear-ephemeral.test.lua, vinyl]
- [sql/gh-3199-no-mem-leaks.test.lua, vinyl]
```
```
- [sql/check-clear-ephemeral.test.lua, memtx]
- [sql/gh-3199-no-mem-leaks.test.lua, vinyl]
```
```
- [sql/check-clear-ephemeral.test.lua, vinyl]
- [sql/gh-3199-no-mem-leaks.test.lua, memtx]
```

**Comment by ImeevMA** [comment]
As far as I understand, this error appeared after [this](https://github.com/tarantool/tarantool/commit/06c73f1dfa34e8b38afe38da67ae24e7a2665eb8) commit

**Comment by Totktonada** [comment]
See also #3732 and https://github.com/tarantool/test-run/issues/131

**Comment by GeorgyKirichenko** [comment]
closed by e4fc22d00d1a9088631fe8eae0d08157834d94d8
